### ISSUE TITLE
Grubs sometimes not resolving

### ISSUE THREAD

**Issue by cs716** [apetavern/grubs#395]
- Grubs will sometimes fail to resolve causing the game to lock up
- This frequently happens when #394 occurs and a grub is launched. Grub potentially not resolving due to continued movement while flying away? 

We could potentially put a band-aid on this by forcing grubs that do not resolve after a reasonable period to suicide.

**Comment by cs716** [comment]
Some unstuck changes may have fixed this. I haven't been able to repro in my testing. Will monitor on PT Friday

**Comment by DrewBritt** [comment]
I believe I fixed the (a?) problem leading to the game freezing; if a Grub was damaged after the turn ended (landmine detonating another landmine is how I reliably repro'd), they'd try to call UseTurn again, which led to the gamemode locking. I've simply told the call to skip if the turn is already changing. I've also added a failsafe for Grubs not resolving, as it was the only ```await GameTask.Delay()``` call in a while loop that had the potential to run forever.

https://github.com/apetavern/grubs/commit/6b7dcbf4ce1845aaf1716f9e43d913c8e597614a
